File: video-gerador/src/services/cutout.py
"""
Recorta o personagem da imagem (remove o fundo), deixando um PNG transparente.
Uso: python3 cutout.py <caminho_entrada> <caminho_saida>
"""
import sys
from rembg import remove
from PIL import Image

def main():
    if len(sys.argv) != 3:
        print("uso: python3 cutout.py <entrada> <saida>", file=sys.stderr)
        sys.exit(1)

    entrada, saida = sys.argv[1], sys.argv[2]
    imagem = Image.open(entrada)
    resultado = remove(imagem)
    resultado.save(saida)
    print(saida)

if __name__ == "__main__":
    main()

File: video-gerador/src/services/cutout.js
const { spawn } = require("child_process");
const path = require("path");

/**
 * Recorta o personagem da imagem de entrada (remove o fundo).
 * @param {string} inputPath  caminho da imagem original
 * @param {string} outputPath caminho de saída (PNG transparente)
 * @returns {Promise<string>} caminho do PNG gerado
 */
function cutoutCharacter(inputPath, outputPath) {
  return new Promise((resolve, reject) => {
    const scriptPath = path.join(__dirname, "cutout.py");
    const proc = spawn("python3", [scriptPath, inputPath, outputPath]);

    let stderr = "";
    proc.stderr.on("data", (chunk) => (stderr += chunk.toString()));

    proc.on("close", (code) => {
      if (code === 0) {
        resolve(outputPath);
      } else {
        reject(new Error(`cutout.py falhou (código ${code}): ${stderr}`));
      }
    });
  });
}

module.exports = { cutoutCharacter };
